### ISSUE TITLE
Add long term memory to think thoroughly agent

### ISSUE THREAD

**Issue by gabrielfior** [gnosis/prediction-market-agent#149]
Currently, the microchain general agent has long term memory (i.e. it writes to the DB).
We should add this functionality to the `think-thoroughly` agent. This will allow the Farcaster agent to have access to its reasoning and thus produce better tweets/casts.

**Comment by gabrielfior** [comment]
Suggestions for grabbing memory from crewai:

```python
# add memory=True to each Agent
 # Log memories  
 summary = researcher.agent_executor.memory.buffer  
 for message in researcher.agent_executor.memory.chat_memory.messages:  
    print(message.content, message.type)  
 .memory.buffer
```

**Comment by kongzii** [comment]
Should it be long-term memory that agent will use for predictions, or just dumps of his reasoning so the Farcaster agent can use it?

**Comment by evangriffiths** [comment]
Yes I think so too.

**Comment by gabrielfior** [comment]
This task is rather trivial if we are able to update crewai 0.22.2 -> 0.27.0 (where memory was first added).
If not, alternative approaches must be considered.

**Comment by gabrielfior** [comment]
Closed by #214
